docs/docs/guides/04-providers/03-email-http-api.md

```diff
@@ -0,0 +1,139 @@
+---
+id: email-http
+title: Using HTTP Email APIs with Auth.js
+---
+
+## Introduction
+
+We have always provided an Email provider with Auth.js with which you could provide to any SMTP server of your choice to send use "magic link" emails to sign-in with. However, as time went on, we had more and more people as us to make it easy to use HTTP-based Email providers, like AWS SES, Postmark, Sendgrid, etc. This was theoretically always possible, but we didn't make it as straight-forward as possible. In this guide I'm going to explain how to use our Email magic link provider with any of the more modern HTTP-based Email APIs.
+
+For this API I will be using [Sendgrid](https://sendgrid.com), but any Email provider which provides an HTTP API / JS-client library will do the trick.
+
+## Setup 
+
+First, clone and setup a basic Auth.js project like the the one [provided in our example repo](https://github.com/nextauthjs/next-auth-example.git). I won't go into the details of setting up your project for Auth.js's [Email provider](https://next-auth.js.org/providers/email), but you will need to make at least the following modifications to the example repository, or any other project you're adding this support to, including:
+
+  - Install `nodemailer` as a dependency. The email provider also requires the use of a database adapter for Auth.js.
+  - Install `@prisma/client` and `@next-auth/prisma-adapter` as dependencies, and `prisma` as a dev dependency and setup Prisma as detailed in [our guide](https://next-auth.js.org/adapters/prisma).
+  - Generate an API key from your cloud Email provider of choice and add it to your `.env.*` file. For example, mine is going to be called `SENDGRID_API`.
+
+At this point, you should have a `[...nextauth].ts` file which looks something like this:
+
+```js title="/src/pages/api/auth/[...nextauth].ts"
+import NextAuth, { NextAuthOptions } from "next-auth"
+import EmailProvider from "next-auth/providers/email"
+import { PrismaAdapter } from "@next-auth/prisma-adapter"
+import prisma from "../../../lib/db"
+
+// For more information on each option (and a full list of options) go to
+// https://next-auth.js.org/configuration/options
+export const authOptions: NextAuthOptions = {
+  // https://next-auth.js.org/configuration/providers/oauth
+  providers: [
+    EmailProvider({
+      server: process.env.EMAIL_SERVER,
+      from: process.env.EMAIL_FROM
+    }),
```

Comment:
these configs are irrelevant for the HTTP-based provider

Reply:
Yeah this is how it starts when folks copy the Email provider out of the docs though/start from any template / starter. So I wanted to make it clear this is the starting point.. in later code blocks we remove those.